tracer/test/Datadog.Trace.Tests/DataStreamsMonitoring/DataStreamsWriterTests.cs

```diff
@@ -21,53 +23,87 @@ namespace Datadog.Trace.Tests.DataStreamsMonitoring;
+    private int _flushCount;
 
-    [Fact]
-    public async Task DoesNotWriteIfNoStats_Periodic()
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    [InlineData(null)]
+    public async Task DoesNotWriteIfNoStats_Periodic(bool? isSupported)
     {
         var bucketDurationMs = 100; // 100 ms
         var api = new StubApi();
-        var writer = CreateWriter(api, bucketDurationMs * 1_000_000);
+        var writer = CreateWriter(api, out var discovery, bucketDurationMs);
+        writer.FlushComplete += (_, _) => Interlocked.Increment(ref _flushCount);
+        TriggerSupportUpdate(discovery, isSupported);
 
-        await Task.Delay(bucketDurationMs * 10);
+        await WaitForFlushCount(bucketDurationMs * 10);
 
         api.Sent.Should().BeEmpty();
     }
 
-    [Fact]
-    public async Task DoesNotWriteIfNoStats_OnClose()
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    [InlineData(null)]
+    public async Task DoesNotWriteIfNoStats_OnClose(bool? isSupported)
     {
         var bucketDuration = 100_000_000; // 100 ms
         var api = new StubApi();
-        var writer = CreateWriter(api, bucketDuration);
+        var writer = CreateWriter(api, out var discovery, bucketDuration);
+        TriggerSupportUpdate(discovery, isSupported);
 
         await writer.DisposeAsync();
 
         api.Sent.Should().BeEmpty();
     }
 
     [Fact]
-    public async Task WritesAStatsPointAfterDelay()
+    public async Task WhenSupported_WritesAStatsPointAfterDelay()
     {
         var bucketDurationMs = 100; // 100 ms
         var api = new StubApi();
-        var writer = CreateWriter(api, bucketDurationMs * 1_000_000);
+        var writer = CreateWriter(api, out var discovery, bucketDurationMs);
+        TriggerSupportUpdate(discovery, isSupported: true);
 
         writer.Add(CreateStatsPoint());
 
-        await Task.Delay(bucketDurationMs * 10);
+        await api.WaitForCount(1, 30_000);
+
+        api.Sent.Should().ContainSingle();
 
         await writer.DisposeAsync();
 
         api.Sent.Should().ContainSingle();
     }
 
+    [Theory]
+    [InlineData(false)]
+    [InlineData(null)]
+    public async Task WhenNotSupported_DoesNotWritesAStatsPointAfterDelay(bool? isSupported)
+    {
+        var bucketDurationMs = 100; // 100 ms
+        var api = new StubApi();
+        var writer = CreateWriter(api, out var discovery, bucketDurationMs * 1_000_000);
+        writer.FlushComplete += (_, _) => Interlocked.Increment(ref _flushCount);
+        TriggerSupportUpdate(discovery, isSupported);
+
+        writer.Add(CreateStatsPoint());
+
+        await WaitForFlushCount(bucketDurationMs * 10);
+
+        await writer.DisposeAsync();
+
+        api.Sent.Should().BeEmpty();
+    }
+
     [Fact]
-    public async Task WritesAStatsPoint_OnClose()
+    public async Task WhenSupported_WritesAStatsPoint_OnClose()
     {
         var bucketDuration = 100_000_000; // 100 ms
```

Comment:
nitpicky (and outside of the PR)
```suggestion
        var bucketDuration = 100_000_000;
```